packages/core/test/template-engine/built-in-extensions/cache-layer/cache.spec.ts

```diff
@@ -0,0 +1,62 @@
+import { createTestCompiler } from '../../testCompiler';
+
+it('If exist arguments behind "cache" symbol, extension should throw an error', async () => {
+  // Arrange
+  const { compiler } = await createTestCompiler();
+
+  // Action, Assert
+  await expect(
+    compiler.compile(`{% cache variable %} some statement {% endcache %}`)
+  ).rejects.toThrow(`Expected a block end, but got symbol`);
+});
+
+it('Extension compiled succeed even if exist multiple "cache" scope tags ', async () => {
+  // Arrange
+  const { compiler } = await createTestCompiler();
+
+  // Action, Assert
+  await expect(
+    compiler.compile(`
+  {% cache %}{% endcache %}
+  {% cache %} some statement {% endcache %}
+    `)
+  ).resolves.not.toThrow();
+});
+
+it('Extension compiled succeed even if not have query statement body in "cache" scope tag', async () => {
+  // Arrange
+  const { compiler } = await createTestCompiler();
+
+  // Action, Assert
+  await expect(
+    compiler.compile(`{% cache %}{% endcache %}`)
+  ).resolves.not.toThrow();
+});
+
+it('The metadata should include"isUsedTag" is true in "cache.vulcan.com" after compiled', async () => {
+  // Arrange
+  const { compiler } = await createTestCompiler();
+
+  // Action
+  const { metadata } = await compiler.compile(
+    `{% cache %} some statement {% endcache %}`
+  );
+
```

Comment:
should we add a test case to indicate that the isUsedTag will be true even if there is no statement body in the cache tag?
``` typescript
const { metadata } = await compiler.compile(
    `{% cache %}{% endcache %}`
  );
await expect(metadata).toMatchObject({
    'cache.vulcan.com': { isUsedTag: true },
  });
```

Reply:
Sure, but I think we don't need to add another test case because it's another sample for testing the `isUsedTag`, so I will use `jest.each` to add the sample it the original test cases